README.md

```diff
@@ -1 +1,218 @@
+```
+KA...
+/Users/azat/Documents/Code/learn-co/node-patterns-concurrency/async-throw.js:2
+  setTimeout(function(){  throw new Error('BOOM!')}, 0)
+                          ^
+
+Error: BOOM!
+    at null._onTimeout (/Users/azat/Documents/Code/learn-co/node-patterns-concurrency/async-throw.js:2:33)
+    at Timer.listOnTimeout (timers.js:92:15)
+```
+
+
+Note: When you `setTimeout` with 0, it will schedule the callback at 0 milliseconds, but on the next tick (same as iteration or cycle) of the event loop. For this reason, Node will finish with all the statements in this file first and only after that on the next cycle, it will fire up the callback.
+
+
+The "KA..." message is shown first, then the delayed callback throws an error which kills the program. When we don't have asynchronous code, `try/catch` works. However, with asynchronous code `try/catch` is useless unless you put it inside of the callback. Why is that?
+
+If we go deeper under the hood, callbacks are scheduled by the event loop. They lose context of the  environment in which they were defined (and lose `try/catch` which is around them). In a way, future code (callbacks) and current code (`try/catch`) become separated by the Node's non-blocking I/O and event loop mechanisms. 
+
+What should we do?
+
+## Async Error Handling
+
+Node developers use error-as-the-first-argument pattern to bubble up the errors. When you have an error which comes from an asynchronous code, check for it. Always check for these errors and pass them up the chain. 
+
+Consider this code in which we have a `callback` function and use it to pass either `null` or `error`:
+
+```js
+module.exports = function(inputFile, customers, callback){
+  return function writeFile(error){
+    if (!error) {
+      return callback(null, customers)
+    } else {
+      return callback(error)
+    }
+  }
+}
+```
+
+The same code can be refactored into more compact version:
+
+```js
+module.exports = function(inputFile, customers, callback){
+  return function writeFile(error){
+    if (error) return callback(error)
+    return callback(null, customers)
+  }
+}
+```
+
+By using `if (error) return callback(error)` you will NOT increase the indentation which leads to callback hell. As you know, `return` will stop the execution and return the value making this function an expression. We don't really care about it being the expression. We use `return` to control the flow.
+
+If you don't have an error (it's falsy), just continue with your logic. At some point, you might want to return some data back to the original caller via the callback. Pass the data as the second argument, leaving the first for the error which is `null` if everything went fine.
+
+Because of the way closures work in JavaScript/Node (they have access to outer scope), you can invoke `callback` at any level when you have multiple levels of nested callbacks:
+
+```js
+var fs = require('fs')
+var inputFile = './customers.csv'
+
+module.exports = function(callback){
```

Comment:
This example is a little difficult to parse for demonstrating the availability of an error callback throughout this function's scope.